tests/jax/components/training/step_test.py

```diff
@@ -0,0 +1,284 @@
+
+    assert mock_trainer.store.trainer_logger.written == {"next_sample": 2, "sample": 1}
+
+
+def test_mapg_with_trust_region_step_initiator() -> None:
+    """Test constructor of MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    assert mapg_with_trust_region_step.config.discount == 0.99
+
+
+def test_on_training_init_start(mock_trainer: MockTrainer) -> None:
+    """Test on_training_init_start method from MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    mapg_with_trust_region_step.on_training_init_start(trainer=mock_trainer)
+
+    assert mock_trainer.store.full_batch_size == 4
+
+
+def test_on_training_step_fn(mock_trainer: MockTrainer) -> None:
+    """Test on_training_step_fn method from MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    del mock_trainer.store.step_fn
+    mapg_with_trust_region_step.on_training_step_fn(trainer=mock_trainer)
+
+    assert callable(mock_trainer.store.step_fn)
+
+
+def test_step(mock_trainer: MockTrainer, dummy_sample: DummySample) -> None:
+    """Test step function"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    del mock_trainer.store.step_fn
+    mapg_with_trust_region_step.on_training_step_fn(trainer=mock_trainer)
+    old_key = mock_trainer.store.key
+    with jax.disable_jit():
+        metrics = mock_trainer.store.step_fn(dummy_sample)
+
+    assert list(metrics.keys()) == [
+        "norm_params",
+        "observations_mean",
+        "observations_std",
+        "rewards_mean",
+        "rewards_std",
+    ]
+
+    assert metrics["norm_params"] == 3.8729835
+
+    assert metrics["observations_mean"] == 0.5667871
+
+    assert metrics["observations_std"] == 0.104980744
+
+    assert list(metrics["rewards_mean"].keys()) == ["agent_0", "agent_1", "agent_2"]
+    assert float(list(metrics["rewards_mean"].values())[0]) == 0.0
+    assert float(list(metrics["rewards_mean"].values())[1]) == 0.08261970430612564
+    assert float(list(metrics["rewards_mean"].values())[2]) == 0.07156813144683838
+
+    assert list(metrics["rewards_std"].keys()) == ["agent_0", "agent_1", "agent_2"]
+    assert float(list(metrics["rewards_std"].values())[0]) == 0.0
+    assert float(list(metrics["rewards_std"].values())[1]) == 0.07048596441745758
+    assert float(list(metrics["rewards_std"].values())[2]) == 0.07740379124879837
+
+    assert list(mock_trainer.store.key) != list(old_key)
+
+    for i, net_key in enumerate(mock_trainer.store.networks["networks"]):
+        assert jnp.array_equal(
+            mock_trainer.store.networks["networks"][net_key].params["key"],
+            jnp.array([i + 2.0, i + 2.0, i + 2.0]),
+        )
+
+    assert mock_trainer.store.opt_states == {
+        "network_agent_0": 2,
+        "network_agent_1": 3,
+        "network_agent_2": 4,
+    }
```

Comment:
Similarly, these are the original values that have been incremented `mock_trainer.store.num_epochs * mock_trainer_store.num_minibatches` times. Is there are way to check this? Then we again verify that step was called the correct number of times.